macro-plantuml-macro/src/main/java/org/xwiki/contrib/plantuml/PlantUMLGenerator.java

```diff
@@ -33,6 +33,27 @@
 @Role
 public interface PlantUMLGenerator
 {
+    /**
+     * Generate the image in the passed output parameter, using PlantUML.
+     *
+     * @param input the textual definition input
+     * @param output the stream into which the generated image will be written to
+     * @param serverURL the optional plantUML server URL (e.g. {@code https://www.plantuml.com/plantuml}. If not an
+     *        empty string or not null then the URL is called to get the generated image. Otherwise PlantUML works in
+     *        embedded mode (requires installation of Graphviz locally for most diagram types, and the {@code
+     *        GRAPHVIZ_DOT} environment variable must be set to point to the path of the GraphViz executable).
+     * @throws IOException when there's a generation or writing error
+     * @deprecated replaced with {@link #outputImage(String, OutputStream, String, PlantUMLDiagramFormat)}
+     */
+    @Deprecated
+    default void outputImage(String input, OutputStream output, String serverURL) throws IOException
+    {
+        // TODO: Scheduled for removal in an upcoming release.
+        // Implementations prior to 2.4 have this method overridden.
+        // New implementations must override {@link #outputImage(String, OutputStream, String, PlantUMLDiagramFormat)}.
+        throw new UnsupportedOperationException("Deprecated method");
```

Comment:
The impl is not good, it should call `outputImage(String input, OutputStream output, String serverURL, PlantUMLDiagramFormat format)` with the format that was used.